### PR TITLE
Logging and table polish.

### PR DESCRIPTION
**Pull request by jacob314**
Switch from reversed table mode to auto-scroll to bottom.
Specify the root of the visibility observer for more more accurate
notification of when content changed.

Strip out reversed table support as it isn't needed.

Make Column slightly more flexible to support creating content
by mechanisms other than innerHtml and text

Hide the Error object from the UI when it doesn't have a meaningful toString.

### CODE REVIEW

**Comment by jacob314** [comment]
screen shot 2019-01-17 at 1 58 26 pm
Cleaner error message when unable to connect.

File: lib/framework/framework.dart [hunk 1]
```diff
@@ -120,7 +120,11 @@ class Framework {
     flash.addClose().click(clearError);
     flash.add(label(text: title));
     if (error != null) {
-      flash.add(div(text: '$error'));
+      final errorString = '$error';
+      // Only display the error object if it has a custom Dart toString.
+      if (!errorString.startsWith('[object ')) {
```
Comment (devoncarew) [inline lib/framework/framework.dart]: 👍

File: lib/logging/logging.dart [hunk 2]
```diff
@@ -321,7 +323,17 @@ class LoggingScreen extends Screen {
...
+      // TODO(jacobr): adding data should be more incremental than this.
+      // We are blowing away state for all already added rows.
       loggingTable.setRows(data);
+      // Smooth scroll if we havent scrolled in a while, otherwise use an
+      // immediate scroll because repeatedly smooth scrolling on the web means
+      // you never reach your destination.
+      final DateTime now = DateTime.now();
+      final bool smoothScroll = _lastScrollTime == null || _lastScrollTime.difference(now).inSeconds > 1;
+      _lastScrollTime = now;
+      DateTime.now().millisecondsSinceEpoch;
```
Comment (devoncarew) [inline lib/logging/logging.dart]: unneeded line?

File: lib/tables.dart [hunk 3]
```diff
@@ -14,15 +14,16 @@ import 'utils.dart';
 
 // TODO(devoncarew): fixed position header
 
+class TableRow {}
```
Comment (devoncarew) [inline lib/tables.dart]: Is this class used?

File: lib/tables.dart [hunk 4]
```diff
@@ -166,6 +166,26 @@ class Table<T> extends Object with SetStateMixin {
...
+    final index = data.indexOf(row);
+    if (index == -1) {
+      return;
+    }
+    if (_hasPendingRebuild) {
+      // Wait for the content to be rendered before we scroll otherwise we may
+      // not be able to scroll far enough.
+      setState(() {
+        // We assume the index is still valid. Alternately we search for the
+        // inex again. The one thing we should absolutely not do is call the
```
Comment (devoncarew) [inline lib/tables.dart]: sp: index

File: lib/logging/logging.dart (hunk 2 above)
Comment (kenzieschmoll) [inline lib/logging/logging.dart]: what is this line doing?

**Pull request merged**